### DISCUSSION TITLE
Is there anybody interested to review existing `secure` feature?

### DISCUSSION THREAD

**Discussion by Oldes** [Oldes/Rebol3#90] [Q&A]
Since the early R3-alpha version when you open the console, there is this notice about not fully available security:

![image](https://github.com/Oldes/Rebol3/assets/454805/57602157-bccf-4ae9-b694-fe9730836a0a)

The true is, that it is available, just not fully tested and reviewed. Is there someone ready to review the existing functionality and let me know, what could be changed/improved so I could remove the notice?

First of all, there should be mentioned `secure` as a special function like `help`. And it should provide some usage examples.

For example it is possible to limit maximum memory of a Rebol process using:
```rebol
secure [memory 1'000'000]
make string! 2'000'000 ;;<- quits, because memory usage is exceeded
```

Quickly checking... it is possible to limit `browse`:
```rebol
>> secure [browse throw]
>> browse http://rebol.com

** Access error: security violation: http://rebol.com  (refer to SECURE function)
```
But not access to `net`:
```rebol
>> secure [net throw]
>> read http://rebol.com
 [HTTP] Opening connection: tcp://rebol.com:80
 ```
 which is one of the missing features.. and I would like to find all :-/

**Comment by Oldes** [comment]
Just for a record, current `secure` help looks like:
```
>> help secure
USAGE:
     SECURE 'policy

DESCRIPTION:
     Set security policies (use SECURE help for more information).
     SECURE is a function! value.

ARGUMENTS:
     'policy       [word! lit-word! block! unset!] Set single or multiple policies (or HELP).

>> secure
 You can set policies for:
     file
     net
     eval
     memory
     secure
     protect
     debug
     envr
     call
     browse
     extension
 These can be set to:
     allow  - no security
     throw  - throw as an error
     quit   - exit the program immediately
     file   - a file path
     url    - a file path
     other  - other value, such as integer
 Settings for read, write, and execute are also available.

```

**Comment by greggirwin** [comment]
I'll keep this open as a reminder. It's a nice feature.

**Comment by rebolek** [comment]
I'm using `secure` for few things, I can take a deeper look.
